CMakeLists.txt

```diff
@@ -128,14 +135,18 @@ if( MIOPEN_BACKEND STREQUAL "OpenCL")
         if(MIOPEN_HIP_COMPILER)
             message("hip compiler: ${MIOPEN_HIP_COMPILER}")
         else()
-            message(FATAL_ERROR "hip compiler not found")
-        endif()
+        message(FATAL_ERROR "hip compiler not found")
+    endif()
     endif()
 
     set (HIP_CXX_COMPILER ${MIOPEN_HIP_COMPILER})
 
     # scgemm not support opencl backend yet.
     set(MIOPEN_USE_SCGEMM OFF)
+
+    # Disable comgr for now for OCL backend.
+    # TODO (priority_low) Use to build HIP and ASM kernels.
+    set(MIOPEN_USE_COMGR Off)
```

Comment:
This should be a different variable name, like `MIOPEN_USE_COMGR_DEFAULT`. And then the option should be moved down and do `option(MIOPEN_USE_COMGR "When possible, use comgr to build kernels instead of offline tools" ${MIOPEN_USE_COMGR_DEFAULT})`.

Reply:
This would leave `MIOPEN_USE_COMGR_DEFAULT` unset for HIP backend. Also this would allow to build OpenCL with `MIOPEN_USE_COMGR=On` if `-DMIOPEN_USE_COMGR=On` is supplied.

Could you please clarify what you would like to achieve? Current behavior:
- `MIOPEN_USE_COMGR` is off by default
- `MIOPEN_USE_COMGR` can be enabled for HIP backend by `-DMIOPEN_USE_COMGR=On`.
- ``MIOPEN_USE_COMGR` cannot be enabled for OCL backend.

Reply:
> Also this would allow to build OpenCL with MIOPEN_USE_COMGR=On if -DMIOPEN_USE_COMGR=On is supplied.

It should be a hard error if the user sets `MIOPEN_USE_COMGR=On` on the opencl backend. We shouldnt silently disable it. This way a user wont think they have enabled it when it doesn't work.

> Could you please clarify what you would like to achieve?

If it is disabled by default then we dont need `MIOPEN_USE_COMGR_DEFAULT`, we just need `option(MIOPEN_USE_COMGR "When possible, use comgr to build kernels instead of offline tools" Off)` and remove all the `set(MIOPEN_USE_COMGR On)` or  `set(MIOPEN_USE_COMGR Off)` in the cmake.

Reply:
This is intermediate stuff, but okay.

Reply:
Done.